scipy/optimize/_qap.py

```diff
@@ -0,0 +1,780 @@
+        variates.
+        If `rng` is ``None`` the `~np.random.RandomState` singleton is
+        used.
+        If `rng` is an int, a new ``RandomState`` instance is used,
+        seeded with `rng`.
+        If `rng` is already a ``RandomState`` or ``Generator``
+        instance, then that object is used.
+        Default is None.
+
+    partial_match : 2d-array of integers, optional, (default = None)
+        Allows the user to fix part of the matching between the two
+        matrices. In the literature, a partial match is also known as a
+        "seed".
+
+        Each row of `partial_match` specifies the indices of a pair of
+        corresponding nodes, that is, node ``partial_match[i, 0]`` of `A` is
+        matched to node ``partial_match[i, 1]`` of `B`. Accordingly,
+        ``partial_match`` is an array of size ``(m , 2)``, where ``m`` is
+        less than the number of nodes.
+
+    init_J : 2d-array or "barycenter" (default = "barycenter")
+        The initial (guess) permutation matrix or search "position"
+        :math:`J`.
+
+        :math:`J` need not be a proper permutation matrix;
+        however, it must have the same shape as `A` and
+        `B`, and it must be doubly stochastic: each of its
+        rows and columns must sum to 1.
+
+        If unspecified or ``"barycenter"``, the non-informative "flat
+        doubly stochastic matrix" :math:`1*1^T/n`, where :math:`n`
+        is the number of nodes and :math:`1` is a :math:`n \times 1`
+        array of ones, is used. This is the "barycenter" of the
+        search space of doubly-stochastic matrices.
+
+    init_weight : float in range [0, 1]
+        Allows the user to specify the weight of the provided
+        search position :math:`J` relative to random perturbations
+        :math:`K`.
+
+        The algorithm will be repeated :math:`k` times
+        from randomized initial search positions
+        :math:`P_0 = (\alpha J + (1- \alpha) K`, where
+        :math:`J` is given by option `init_J`,
+        :math:`\alpha` is given by option `init_weight`,
+        :math:`k` is given by option `init_k`, and
+        :math:`K` is a random doubly stochastic matrix.
+
+        Default is 1 if `init_k` is 1, 0 otherwise.
+    init_k : int, positive (default = 1)
+        The number of randomized initial search positions :math:`P_0`
+        from which the FAQ algorithm will proceed.
+    maxiter : int, positive (default = 30)
+        Integer specifying the max number of Franke-Wolfe iterations
+        per initial search position. FAQ typically converges within a
+        modest number of iterations.
+    shuffle_input : bool (default = True)
+        To avoid artificially high or low matching due to inherent
+        sorting of input matrices, gives users the option
+        to shuffle the nodes. Results are then unshuffled so that the
+        returned results correspond with the node order of inputs.
+    eps : float (default = 0.05)
```

Comment:
I think we should rename this `tol` for consistency with .optimize.minimize`.

Reply:
`eps` is used as a step size in e.g. BFGS

Reply:
Done.